osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -196,13 +209,55 @@ public override bool OnBackButton()
 
         private void onPlaylistChanged(object sender, NotifyCollectionChangedEventArgs e) => SelectedItem.Value = Playlist.FirstOrDefault();
 
+        private void onReadyClick()
+        {
+            Debug.Assert(readyClickOperation == null);
+            readyClickOperation = ongoingOperationTracker.BeginOperation();
+
+            if (client.IsHost && client.LocalUser?.State == MultiplayerUserState.Ready)
+            {
+                client.StartMatch()
+                      .ContinueWith(t =>
+                      {
+                          // accessing Exception here silences any potential errors from the antecedent task
+                          if (t.Exception != null)
+                          {
+                              t.CatchUnobservedExceptions(true); // will run immediately.
+                              // gameplay was not started due to an exception; unblock button.
+                              endOperation();
+                          }
+
+                          // gameplay is starting, the button will be unblocked on load requested.
+                      });
+                return;
+            }
+
+            client.ToggleReady()
+                  .ContinueWith(t =>
+                  {
+                      t.CatchUnobservedExceptions(true); // will run immediately.
+                      endOperation();
+                  });
+
+            void endOperation()
+            {
+                readyClickOperation?.Dispose();
```

Comment:
I think a non-null assert here could work too, if you want to go crazy with assertions.

Reply:
Right. This is a vestige of an earlier revision in which I had the tracker as `CanBeNull` (but later realised that it does not need to be).